components/motor/motor.go

```diff
@@ -35,12 +35,66 @@ const SubtypeName = "motor"
 var API = resource.APINamespaceRDK.WithComponentType(SubtypeName)
 
 // A Motor represents a physical motor connected to a board.
+// 
+// SetPower example:
+//    myMotorComponent, err := motor.FromRobot(machine, "my_motor")
+//    // Set the motor power to 40% forwards.
+//    myMotorComponent.SetPower(context.Background(), 0.4, nil)
+//
+// GoFor example:
+//    myMotorComponent, err := motor.FromRobot(machine, "my_motor")
+//    // Turn the motor 7.2 revolutions at 60 RPM.
+//    myMotorComponent.GoFor(context.Background(), 60, 7.2, nil)
+//
+// GoTo example:
+//    // Turn the motor to 8.3 revolutions from home at 75 RPM.
+//    myMotorComponent.GoTo(context.Background(), 75, 8.3, nil)
+//
+// ResetZeroPostion example:
+//    // Set the current position as the new home position with no offset.
+//    myMotorComponent.ResetZeroPosition(context.Background(), 0.0, nil)
+//
+// Position example:
+//    // Get the current position of the motor.
```

Comment:
Discussion question: `Position` and `ResetZeroPosition` only make sense for motors _with encoders._ Is it worth mentioning that here? Maybe not; it's surely mentioned elsewhere already.

Reply:
Ah great point. The docs do mention something about encoders for Position, but now that you raise it I'm clarifying it for both of these methods on the docs page. Here though, idk--should it be more streamlined, with more info at docs.viam.com, or should this have all the long-winded info? I'm really not sure. Here would be a compromise:
```suggestion
//    // Get the current position of an encoded motor.
```

Reply:
I like the compromise: it has the important part, and if that confuses anyone they should know to look up the real docs.

Reply:
okay! ooh i'll add it to the comments for this and resetzero in the actual interface comments down below